tests/components/template/test_sensor.py

```diff
@@ -15,443 +15,393 @@
+        assert await async_setup_component(
+            hass,
+            sensor.DOMAIN,
+            {
+                "sensor": {
+                    "platform": "template",
+                    "sensors": {
+                        "test_template_sensor": {
+                            "value_template": "{{ states.sensor.test_state.state }}",
+                            "attribute_templates": {
+                                "test_attribute": "It {{ states.sensor.test_state.state }}."
                             },
-                        },
-                    }
-                },
-            )
-        self.hass.block_till_done()
-
-        state = self.hass.states.get("sensor.test1")
-        assert state.attributes["device_class"] == "temperature"
-        state = self.hass.states.get("sensor.test2")
-        assert "device_class" not in state.attributes
-
-    def test_available_template_with_entities(self):
-        """Test availability tempalates with values from other entities."""
-
-        with assert_setup_component(1):
-            assert setup_component(
-                self.hass,
-                "sensor",
-                {
-                    "sensor": {
-                        "platform": "template",
-                        "sensors": {
-                            "test_template_sensor": {
-                                "value_template": "{{ states.sensor.test_state.state }}",
-                                "availability_template": "{{ is_state('availability_boolean.state', 'on') }}",
-                            }
-                        },
-                    }
-                },
-            )
+                        }
+                    },
+                }
+            },
+        )
+
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.test_template_sensor")
+    assert state.attributes.get("test_attribute") == "It ."
+
+    hass.states.async_set("sensor.test_state", "Works")
+    await hass.async_block_till_done()
+    state = hass.states.get("sensor.test_template_sensor")
+    assert state.attributes["test_attribute"] == "It Works."
+
+
+async def test_template_syntax_error(hass):
+    """Test templating syntax error."""
+    with assert_setup_component(0, sensor.DOMAIN):
+        assert await async_setup_component(
+            hass,
+            sensor.DOMAIN,
+            {
+                "sensor": {
+                    "platform": "template",
+                    "sensors": {
+                        "test_template_sensor": {"value_template": "{% if rubbish %}"}
+                    },
+                }
+            },
+        )
 
-        self.hass.block_till_done()
-        self.hass.start()
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+    assert hass.states.async_all() == []
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)